app/src/test/java/org/simple/clinic/scanid/ScanSimpleIdUpdateTest.kt

```diff
@@ -36,6 +51,50 @@ class ScanSimpleIdUpdateTest {
         ))
   }
 
+  @Test
+  fun `when the qr code is scanned and the returned string is json then parse the json into object`() {
+    spec
+        .given(defaultModel)
+        .whenEvent(ScanSimpleIdScreenQrCodeScanned(expectedJson))
+        .then(assertThatNext(
+            hasModel(defaultModel.searching()),
+            hasEffects(ParseJsonIntoPatientPrefillInfoObject(expectedJson))
+        ))
+  }
+
+  @Test
+  fun `when json is parsed then search patient with NHID`() {
+    val patientPrefillInfo: PatientPrefillInfo? = adapter.fromJson(expectedJson)
```

Comment:
We should not be doing that here, instead, we can construct the parsed JSON object.

Reply:
Instead create a function with testdata in `TestData`.